core/src/env.rs

```diff
@@ -184,22 +186,90 @@ impl<E: Ext> LaterExt<E> {
 
         res
     }
+}
+
+/// Struct for interacting with Ext.
+///
+/// Unlike [`BaseExtCarrier`] this struct is intended for external usage.
+pub struct ExtCarrier<E: Ext>(BaseExtCarrier<E>);
+
+impl<E: Ext> ExtCarrier<E> {
+    /// New ext carrier
+    pub fn new(e: E) -> Self {
+        Self(BaseExtCarrier::new(e))
+    }
+
+    /// Unwraps hidden `E` value
+    pub fn into_inner(self) -> E {
+        let BaseExtCarrier { inner } = self.0;
+        inner
+            .take()
+            .expect("can be called only once during instance consumption; qed")
+    }
+}
+
+impl<E: Ext> Deref for ExtCarrier<E> {
+    type Target = BaseExtCarrier<E>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// "Clone-able" struct for interacting with Ext.
+///
+/// Unlike [`BaseExtCarrier`] this struct is intended for external usage.
+// TODO #852 type will be redundant after resolving the issue.
+pub struct ReplicableExtCarrier<E: Ext>(BaseExtCarrier<E>);
```

Comment:
Why don't we name it as is - `ClonableExtCarrier`?

Reply:
because there were no such words **Cloneable**/**Clonable**

Reply:
btw, I am preparing a commit to make it safer using carriers with clone, which leads to removing `ReplicableExtCarrier`

Reply:
Library devs use word `Cloneable` so I guess it's acceptable